go/saltpack/decrypt.go

```diff
@@ -257,13 +258,19 @@ func (ds *decryptStream) processEncryptionBlock(bl *EncryptionBlock) ([]byte, er
 	}
 
 	nonce := ds.nonce.ForPayloadBox(blockNum)
+	ciphertext := bl.PayloadCiphertext
+	hash := sha512.Sum512(ciphertext)
 
-	tag, err := ds.tagKey.Unbox(nonce, bl.TagCiphertexts[ds.position])
+	hashBox, err := ds.tagKey.Box(nonce, hash[:])
 	if err != nil {
+		return nil, err
```

Comment:
I'm looking at `naclBoxSecretKey.Box()`, and that function never returns an error. Neither does the test implementation. Can we delete the error value from the interface here?